### ISSUE TITLE
False Negative: IM_BAD_CHECK_FOR_ODD fails to detect Yoda condition style comparison(1 == i % 2)

### ISSUE THREAD

**Issue by leemeii** [spotbugs/spotbugs#3886]
**Describe the bug**
The rule  **IM_BAD_CHECK_FOR_ODD**correctly identifies the potential issue in the standard comparison i % 2 == 1, but it fails to flag the logically equivalent "Yoda condition" style comparison 1 == i % 2. This leads to a false negative where dangerous code (which fails for negative odd numbers) remains undetected.

**To Reproduce**
To Reproduce Steps to reproduce the behavior:

1.Use the following minimal Java code:

```java

public class TestModulo {
    // Case 1
    public void standardCheck(int i) {
        if (i % 2 == 1) {  // <- triggered
            System.out.println("Odd");
        }
    }

    // Case 2
    public void yodaCheck(int i) {
        if (1 == i % 2) {  // <- should report (FN)
            System.out.println("Odd");
        }
    }
}
```
2.Compile the code: javac TestModulo.java

3.Run SpotBugs: java -jar spotbugs.jar -textui -low -effort:max TestModulo.class

4.Observe that only standardCheck is reported.

**Expected behavior**
Both i % 2 == 1 and 1 == i % 2 should trigger the IM_BAD_CHECK_FOR_ODD warning, as both will fail to return true for negative odd integers.

**Actual behavior**
Only the **case1** method triggers the bug report. **Case2** method is ignored.

**Environment**
- Java version: JDK 21
- SpotBugs Version:  4.9.8
- How do you use SpotBugsCLI
